server/src/main/java/com/cloud/api/ApiSerializerHelper.java

```diff
@@ -30,38 +32,51 @@
 public class ApiSerializerHelper {
     public static final Logger s_logger = Logger.getLogger(ApiSerializerHelper.class.getName());
     private static String token = "/";
+    private static String[] apiPackages = {"com.cloud.agent.api", "org.apache.cloudstack.api"};
```

Comment:
I am not sure if this will improve security (is it the goal of this PR ?).
anyone can create java classes in the packages

Reply:
yes, it is just an extra layer. They will also have to get those classes on the classpath of the remote machine though. I was think of extending the method to also be able to add a required base class or interface.

Reply:
> I am not sure if this will improve security (is it the goal of this PR ?). anyone can create java classes in the packages

any sugestions for improvement @weizhouapache ?